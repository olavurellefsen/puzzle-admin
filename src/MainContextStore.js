import React from "react";
import MainContext from "./Context";

export default function MainContextStore(props) {
  return (
    <MainContext.Provider value={{currentScene: 1, currentPuzzle: 2}}>{props.children}</MainContext.Provider>
  );
}
