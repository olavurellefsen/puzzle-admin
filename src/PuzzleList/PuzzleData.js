export default {
  "data": {
    "puzzle": [
      {
        "id": 1,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": 24,
        "character": "Bina",
        "summary": "Bina velur um hon etur breyð ella havragrýn",
        "sequence": 1
      },
      {
        "id": 2,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": 24,
        "character": "Broddi",
        "summary": "Broddi velur, um hann etur breyð ella havragrýn",
        "sequence": 2
      },
      {
        "id": 3,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 4,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 4
      },
      {
        "id": 5,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 5
      },
      {
        "id": 6,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 6
      },
      {
        "id": 7,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 7
      },
      {
        "id": 8,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 8
      },
      {
        "id": 9,
        "scene_id": 1,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 9
      },
      {
        "id": 10,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 11,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 12,
        "scene_id": 2,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 13,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 14,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 15,
        "scene_id": 3,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 16,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 17,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 18,
        "scene_id": 4,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 19,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 20,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 21,
        "scene_id": 5,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 22,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 23,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 24,
        "scene_id": 6,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 25,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 26,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 27,
        "scene_id": 7,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 28,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 29,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 30,
        "scene_id": 8,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      },
      {
        "id": 31,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 1
      },
      {
        "id": 32,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 2
      },
      {
        "id": 33,
        "scene_id": 9,
        "puzzle_identifier": null,
        "intro_audiofile": null,
        "intro_text_id": null,
        "character": null,
        "summary": null,
        "sequence": 3
      }
    ]
  }
}