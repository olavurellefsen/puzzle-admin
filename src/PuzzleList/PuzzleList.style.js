import styled from "styled-components";

export const PuzzleListContainer = styled.div`
  min-width: 494px;
  overflow: hidden;
`;
PuzzleListContainer.displayName = "PuzzleListContainer";

export const PuzzleListBox = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  padding-right: 14px;
`;
PuzzleListContainer.displayName = "PuzzleListContainer";

