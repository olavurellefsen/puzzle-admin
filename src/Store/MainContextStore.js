import React, { useReducer } from "react";
import MainContext from "./Context";
import createReducer from "./createReducer";

export default (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <MainContext.Provider value={[state, dispatch]}>
      {props.children}
    </MainContext.Provider>
  );
}

const initialState = {
  currentPuzzleitem: 1,
};

const reducer = createReducer(initialState, {
  reset: () => initialState,
  setCurrentPuzzleitem: (state, action) => ({
    ...state,
    currentPuzzleitem: action.number
  }),
});
