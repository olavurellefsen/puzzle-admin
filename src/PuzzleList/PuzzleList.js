import React, { useContext } from "react";
import MainContext from "../Context";
import Puzzle from "./Puzzle/Puzzle";
import { PuzzleListContainer, PuzzleListBox } from "./PuzzleList.style";
import puzzleData from "./PuzzleData";

export default function PuzzleList() {
  const puzzles = puzzleData.data.puzzle;
  const [state, dispatch] = useContext(MainContext);
  let currentPuzzleId = 0;
  return (
    <PuzzleListContainer>
      <PuzzleListBox>
        {puzzles
          .filter(puzzle => puzzle.scene_id === state.currentScene)
          .map((puzzle, index) => {
            if (currentPuzzleId === 0) {
              currentPuzzleId = getCurrentPuzzleId(
                state,
                dispatch,
                puzzle.id,
                state.currentScene
              );
            }
            return (
              <Puzzle
                key={index}
                puzzle={puzzle}
                currentPuzzleId={currentPuzzleId}
              />
            );
          })}
      </PuzzleListBox>
    </PuzzleListContainer>
  );
}

const getCurrentPuzzleId = (state, dispatch, id, scene_id) => {
  let currentPuzzle = state.currentPuzzles.filter(
    p => p.scene_id === state.currentScene
  );
  if (currentPuzzle.length > 0) {
    return currentPuzzle[0].puzzle_id;
  } else {
    // No puzzle selected for the current scene, so set the current one as the selected puzzle
    dispatch({ type: "setCurrentPuzzle", puzzle_id: id, scene_id: scene_id });
    return id;
  }
};
