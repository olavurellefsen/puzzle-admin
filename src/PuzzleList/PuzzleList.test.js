import React from "react";
import { shallow, mount } from "enzyme";
import MainContextStore from "../MainContextStore";
import PuzzleList from "./PuzzleList";
import { ApolloProvider } from "react-apollo";
import client from "../heyBreydLocalizationClient";

describe("PuzzleList", () => {
  let wrapper;
  beforeEach(
    () =>
      (wrapper = shallow(
        <MainContextStore>
          <PuzzleList />
        </MainContextStore>
      ))
  );
  it("should render correctly", () => expect(wrapper).toMatchSnapshot());
});

describe("mounted PuzzleList", () => {
  let wrapper;
  beforeEach(
    () =>
      (wrapper = mount(
        <ApolloProvider client={client}>
          <MainContextStore>
            <PuzzleList />
          </MainContextStore>
        </ApolloProvider>
      ))
  );

  it("should render a PuzzleListContainer", () => {
    expect(wrapper.find("PuzzleListContainer").length).toEqual(1);
  });
});
