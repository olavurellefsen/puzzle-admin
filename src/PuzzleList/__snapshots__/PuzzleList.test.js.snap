// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bZQynM gSqcYA"
    data-testid="PuzzleListContainer"
  >
    <div
      class="sc-gzVnrw ejfrwY"
    >
      G√ÅTUR
    </div>
    <div
      class="sc-htoDjs jRAwCr"
    >
      Loading...
    </div>
  </div>
</div>
`;
